test/Dialect/FIRRTL/imconstprop-aggregate.mlir

```diff
@@ -0,0 +1,447 @@
+// RUN: circt-opt -pass-pipeline='firrtl.circuit(firrtl-imconstprop)' --split-input-file  %s | FileCheck %s
+
+firrtl.circuit "VectorPropagation1" {
+  // CHECK-LABEL: @VectorPropagation1
+  firrtl.module @VectorPropagation1(in %clock: !firrtl.clock, out %b: !firrtl.uint<1>) {
+    %c1_ui1 = firrtl.constant 1 : !firrtl.uint<1>
+    %tmp = firrtl.reg %clock  : !firrtl.vector<uint<1>, 2>
+    %0 = firrtl.subindex %tmp[0] : !firrtl.vector<uint<1>, 2>
+    firrtl.connect %0, %c1_ui1 : !firrtl.uint<1>, !firrtl.uint<1>
+    %1 = firrtl.subindex %tmp[1] : !firrtl.vector<uint<1>, 2>
+    firrtl.connect %1, %c1_ui1 : !firrtl.uint<1>, !firrtl.uint<1>
+    %2 = firrtl.xor %0, %1 : (!firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<1>
```

Comment:
both the read and the write derive from the same subindex op.  A more thorough test would have them derive from different subindex ops so that you were sure the lattice was propagated correctly through the connect.

Reply:
That makes sense. Thanks.